web/src/ChatPage.js

```diff
@@ -275,17 +276,27 @@ class ChatPage extends BaseListPage {
             }
 
             MessageBackend.getMessageAnswer(lastMessage.owner, lastMessage.name, (data) => {
-              if (!chat || (this.state.chat.name !== chat.name)) {
-                return;
-              }
               const jsonData = JSON.parse(data);
 
               if (jsonData.text === "") {
                 jsonData.text = "\n";
               }
               const lastMessage2 = Setting.deepCopy(lastMessage);
               text += jsonData.text;
-              lastMessage2.text = Setting.parseAnswerAndSuggestions(text)["answer"];
+              const parsedResult = parseAnswerWithCarriers(text, 3, !chat.hasTitle);
+              if (parsedResult.title !== "") {
+                const updatedChats = [...this.state.data];
+                const index = updatedChats.findIndex(c => c.name === chat.name);
+                if (index !== -1) {
+                  updatedChats[index].hasTitle = true;
+                  updatedChats[index].displayName = parsedResult.title;
+                  this.setState({data: updatedChats});
+                }
+              }
+              if (!chat || (this.state.chat.name !== chat.name)) {
```

Comment:
Move `if (!chat || (this.state.chat.name !== chat.name)) {
                return;
              }` back to Line 279